src/server/middleware/middleware.go

```diff
@@ -29,3 +29,33 @@ func WithMiddlewares(handler http.Handler, middlewares ...Middleware) http.Handl
 	}
 	return handler
 }
+
+// Skipper defines a function to skip middleware.
+// Returning true skips processing the middleware.
+type Skipper func(*http.Request) bool
+
+// New make a middleware from fn which type is func(w http.ResponseWriter, r *http.Request, next http.Handler)
+func New(fn func(http.ResponseWriter, *http.Request, http.Handler), skippers ...Skipper) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		var f func(w http.ResponseWriter, r *http.Request)
+
+		if len(skippers) == 0 {
```

Comment:
What's the point of this `if`?
You can always call a for loop when skippers is empty it still works as expected?